### PR TITLE
Bugfix/s3 c 3516 fix authz for account lvl metrics

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

File: libV2/vault.js [hunk 1]
```diff
@@ -83,14 +84,29 @@ class Vault {
                         reject(err);
                         return;
                     }
-                    resolve(res);
+                    if (!res.message || !res.message.body) {
+                        reject(errors.InternalError);
```
Comment (miniscruff) [inline libV2/vault.js:88]: should this return or an else for resolve? I don't think promises allow resolve or reject more than once though.

File: libV2/vault.js [hunk 2]
```diff
@@ -83,14 +84,29 @@ class Vault {
...
     }
 }
 
 const vault = new Vault(config);
 auth.setHandler(vault);
 
+async function translateResourceIds(level, resources) {
+    if (level === 'accounts') {
+        const res = await vault.getCanonicalIds(resources);
+        if (res.message || !res.message.body) {
```
Comment (miniscruff) [inline libV2/vault.js]: is this supposed to be !res.message?

**Review by alexanderchan-scality — APPROVED** [review]

File: libV2/vault.js (hunk 1 above)
Comment (alexanderchan-scality) [inline libV2/vault.js:88]:
```suggestion
                        reject(errors.InternalError);
                        return;
```

Comment (tmacro) [reply libV2/vault.js:88]: yep missing a return

**Review by miniscruff — APPROVED** [review]

File: libV2/vault.js [hunk 3]
```diff
@@ -83,14 +84,30 @@ class Vault {
...
                 }));
     }
 }
 
 const vault = new Vault(config);
 auth.setHandler(vault);
 
+async function translateResourceIds(level, resources, log) {
+    if (level === 'accounts') {
+        const res = await vault.getCanonicalIds(resources, log);
```
Comment (jonathan-gramain) [inline libV2/vault.js]: minor style: you could `return await ...`

**Review by jonathan-gramain — APPROVED** [review]

**Pull request merged**